internal/clients/team4/president.go

```diff
@@ -0,0 +1,351 @@
+				taxAmountMap[clientID] = shared.Resources(0)
+			} else if clientReport.ReportedAmount*1.5 <= p.parent.ServerReadHandle.GetGameConfig().CostOfLiving {
+				taxAmountMap[clientID] = shared.Resources(0)
+			}
+			taxAmountMap[clientID] = p.generateTaxAmount(clientID, clientReport.ReportedAmount)
+		} else {
+			taxAmountMap[clientID] = 15 //flat tax rate
+		}
+
+		//Excuse ourselves if we are not fair
+		if clientID == shared.Team4 {
+			taxAmountMap[clientID] = shared.Resources(float64(taxAmountMap[clientID]) * p.parent.internalParam.fairness)
+		}
+	}
+
+	return shared.PresidentReturnContent{
+		ContentType: shared.PresidentTaxation,
+		ResourceMap: taxAmountMap,
+		ActionTaken: true,
+	}
+}
+
+func (p *president) generateWeAreRichThreshold() (shared.Resources, bool) {
+	disasterPrediction := p.parent.obs.iifoObs.finalDisasterPrediction
+	if p.parent.ServerReadHandle.GetGameState().Season == 1 {
+		return 0, false
+	} else {
+		return shared.Resources(disasterPrediction.Magnitude * 1.5 * disasterPrediction.Confidence), true
+	}
+}
+
+func (p *president) generateTaxAmount(clientID shared.ClientID, clientReport shared.Resources) shared.Resources {
+	if p.parent.ServerReadHandle.GetGameState().Season == 1 {
+		//Static tax for when there is little information
+		return 0.2 * clientReport
+	}
+	disasterPrediction := p.parent.obs.iifoObs.finalDisasterPrediction
+	var averageTax shared.Resources
+	if disasterPrediction.TimeLeft == 0 {
+		averageTax = shared.Resources((disasterPrediction.Magnitude -
+			float64(p.parent.ServerReadHandle.GetGameState().CommonPool)) /
+			p.numIslandsAlive())
+	} else {
+		averageTax = shared.Resources((disasterPrediction.Magnitude -
+			float64(p.parent.ServerReadHandle.GetGameState().CommonPool)) /
+			(p.numIslandsAlive()) * float64(disasterPrediction.TimeLeft))
+	}
+	return averageTax
+
+}
+
+// PaySpeaker pays the speaker a salary.
+func (p *president) PaySpeaker() shared.PresidentReturnContent {
+	SpeakerSalaryRule, ok := p.GameState.RulesInfo.CurrentRulesInPlay["salary_cycle_speaker"]
+	var SpeakerSalary shared.Resources = 0
+	if ok {
+		SpeakerSalary = shared.Resources(SpeakerSalaryRule.ApplicableMatrix.At(0, 1))
+	}
+	return shared.PresidentReturnContent{
+		ContentType:   shared.PresidentSpeakerSalary,
+		SpeakerSalary: SpeakerSalary,
+		ActionTaken:   true,
+	}
+}
+
+// CallSpeakerElection is called by the executive to decide on power-transfer
+func (p *president) CallSpeakerElection(monitoring shared.MonitorResult, turnsInPower int, allIslands []shared.ClientID) shared.ElectionSettings {
+	//Compliance with budget rule
+	_, budgetRuleInPay := p.GameState.RulesInfo.CurrentRulesInPlay["president_over_budget"]
+	actionOverBudget := p.parent.ServerReadHandle.GetGameState().IIGORolesBudget[shared.President]-
+		p.parent.ServerReadHandle.GetGameConfig().IIGOClientConfig.AppointNextSpeakerActionCost < 0
```

Comment:
Maybe add brackets to make this more readable? Looks like your taking the budget minus a bool